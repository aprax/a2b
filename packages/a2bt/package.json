{
  "name": "a2bt",
  "version": "3.0.1",
  "description": "A CLI tool and node library for visualizing binary trees",
  "main": "dist/index.js",
  "bin": "dist/cli.js",
  "types": "dist/index.d.ts",
  "scripts": {
    "start": "node dist/cli.js",
    "build": "tsc",
    "test": "yarn jest --coverage",
    "lint": "yarn prettier --write . && yarn eslint src/ --fix"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/aprax/a2b.git"
  },
  "author": "Adam Prax",
  "license": "MIT",
  "devDependencies": {
    "@typescript-eslint/eslint-plugin": "^5.17.0",
    "@typescript-eslint/parser": "^5.17.0",
    "eslint": "^8.12.0",
    "eslint-config-google": "^0.14.0",
    "eslint-config-prettier": "^8.5.0",
    "eslint-plugin-jest": "^26.1.3",
    "jest": "^27.5.1",
    "prettier": "^2.6.1",
    "ts-jest": "^27.1.4",
    "typescript": "^4.6.3"
  }
}
