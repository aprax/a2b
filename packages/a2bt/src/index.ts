import printNodes from "./printNodes";
import type { Args, Node } from "./printNodes";
import toConsole from "./toConsole";
import treeNodeToBinaryArray from "./treeNodeToBinaryArray";
import toBinarySearchTree from "./toBinarySearchTree";

/**
 * @description  Prints a binary tree to the console
 * @param nodes  The array of nodes to be printed as a binary tree
 * @param args  The optional arguments to be passed to printNodes
 * @return  The output of printNodes to the console
 */
const a2bt = (nodes: Node[], args?: Args) => toConsole(printNodes(nodes, args));
export default a2bt;
export { toConsole, printNodes, treeNodeToBinaryArray, toBinarySearchTree };
