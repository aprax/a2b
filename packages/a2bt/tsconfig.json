{
  "compilerOptions": {
    /* Visit https://aka.ms/tsconfig.json to read more about this file */
    "target": "ES2019",
    "module": "CommonJS",
    "rootDir": "./src",
    "moduleResolution": "Node",
    "outDir": "./dist",
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "strict": true,
    "lib": ["es2021"],
    "skipLibCheck": true,
    "alwaysStrict": true,
    "allowUnusedLabels": false,
    "allowUnreachableCode": false,
    "exactOptionalPropertyTypes": true,
    "noFallthroughCasesInSwitch": true,
    "noImplicitOverride": true,
    "noImplicitReturns": true,
    "noPropertyAccessFromIndexSignature": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "declaration": true,
    "declarationDir": "./dist",
    "sourceMap": true
  },
  "exclude": ["node_modules", "jest.config.ts", "**/*.spec.ts", "dist"]
}
