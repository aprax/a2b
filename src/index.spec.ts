import a2bt from ".";
import type { Args } from "./printNodes";

it("a2bt should be exported", () => {
  const nodes: Args["nodes"] = [1, 2, 3];
  jest.spyOn(console, "log").mockImplementation();

  a2bt({ nodes });
  expect(console.log).toHaveBeenCalled();
});
