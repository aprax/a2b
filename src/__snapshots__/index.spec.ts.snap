// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`printNodes should be exported correctly 1`] = `
Array [
  Array [
    " ",
    " ",
    " ",
    "1",
    " ",
    " ",
    " ",
  ],
  Array [
    " ",
    " ",
    "/",
    " ",
    " ",
    " ",
    " ",
  ],
  Array [
    " ",
    "/",
    " ",
    " ",
    " ",
    " ",
    " ",
  ],
  Array [
    "2",
    " ",
    " ",
    " ",
    " ",
    " ",
    " ",
  ],
]
`;
